python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py

```diff
@@ -2,67 +2,91 @@
 
 
 from logging import Logger
-from typing import Optional
+from typing import Dict, Optional
 
-from semantic_kernel.connectors.ai.open_ai.services.open_ai_chat_completion import (
-    OpenAIChatCompletion,
+from openai.lib.azure import AsyncAzureADTokenProvider
+
+from semantic_kernel.connectors.ai.open_ai.services.azure_config_base import (
+    AzureOpenAIConfigBase,
+)
+from semantic_kernel.connectors.ai.open_ai.services.open_ai_chat_completion_base import (
+    OpenAIChatCompletionBase,
+)
+from semantic_kernel.connectors.ai.open_ai.services.open_ai_handler import (
+    OpenAIModelTypes,
+)
+from semantic_kernel.connectors.ai.open_ai.services.open_ai_text_completion_base import (
+    OpenAITextCompletionBase,
 )
 
 
-class AzureChatCompletion(OpenAIChatCompletion):
-    _endpoint: str
-    _api_version: str
-    _api_type: str
+class AzureChatCompletion(
+    AzureOpenAIConfigBase, OpenAIChatCompletionBase, OpenAITextCompletionBase
+):
+    """Azure Chat completion class."""
 
     def __init__(
         self,
         deployment_name: str,
-        endpoint: Optional[str] = None,
+        endpoint: str,
+        api_version: str = "2022-12-01",
         api_key: Optional[str] = None,
-        api_version: str = "2023-03-15-preview",
+        ad_token: Optional[str] = None,
```

Comment:
Should this be updated to reflect that it's now called "Microsoft Entra" and not "AAD?" Similar for the comment below.